tools/testing/tests/fido2.py

```diff
@@ -34,49 +34,48 @@ def VerifyAttestation(attest, data):
     verifier().verify(attest.att_statement, attest.auth_data, data.hash)
 
 
+def cbor_key_to_representative(key):
+    if isinstance(key, int):
+        if key >= 0:
+            return (0, key)
+        return (1, -key)
+    elif isinstance(key, bytes):
+        return (2, key)
+    elif isinstance(key, str):
+        return (3, key)
+    else:
+        raise ValueError(key)
+
+
+def cmp_cbor_keys(a, b):
+    a = cbor_key_to_representative(a)
+    b = cbor_key_to_representative(b)
+    if a[0] != b[0]:
+        return a[0] - b[0]
+    return not ((a[1] > b[1]) - (a[1] < b[1]))
```

Comment:
Sorry, a thought just occurred and, testing in Python, the default string compare is left-aligned, not right-aligned as CTAP2 needs. I.e. ('b' < 'aa') is false in Python. So both strings and bytes would need to compare first on lengths.

Reply:
Good point, Python string comparison isn't sufficient for this so I rewrote it and added some tests.  Am I correct with this ordering?

```python
cbor_key_list_sorted = [
    0,
    1,
    1,
    2,
    3,
    -1,
    -2,
    "b",
    "c",
    "aa",
    "aaa",
    "aab",
    "baa",
    "bbb",
]
```

Reply:
Yes, I think that ordering is correct.